docs/kubernetes/overlays.md

```diff
@@ -0,0 +1,125 @@
+
+Example:
+`stable-1-19` [overlays](../../deploy/kubernetes/overlays/stable-1-19) bundle can be used to deploy all the components of the driver on kubernetes 1.19.
+`prow-gke-staging-release-rc-1-19` [overlays](../../deploy/kubernetes/overlays/prow-gke-staging-release-rc-1-19) bundle can be used to deploy all the components of the driver on kubernetes 1.19.
+
+## General guidelines for creation/modification of new/existing overlays
+
+When it comes to creation of overlays there are no strict rules on how overlays must be created/modified but following captures some of the common scenarios of adding new capabilities to the CSI driver bundle and how it would affect the overlays creation/modification.
+
+1. Sidecar image version update for existing overlays
+   * Say we have three existing stable overlays `stable-1-17`, `stable-1-18`, `stable-master`. Sidecar S1 is available 1.17+, S2 is avialable 1.18+. If we need to upgrade the image versions for these sidecars, we would do the following:
+   * Update the `deploy/kubernetes/images/prow-gke-staging-release-rc-1-17` for sidecar S1, `deploy/kubernetes/images/prow-gke-staging-release-rc-1-18` and `deploy/kubernetes/images/prow-gke-staging-release-rc-master` for S1 and S2.
+   * Validate the changes
+   * Now update  `deploy/kubernetes/images/stable-1-17` for sidecar S1, and `deploy/kubernetes/images/stable-1-18`, `deploy/kubernetes/images/stable-master` for sidecar S1 and S2.
+
+2. Creating a new k8s minor version overlay with an existing released driver image
+   * Say we have existing overlays `stable-1-18`, `stable-master`. We plan to now create a new overlay for k8s 1.19
+   * Create a new `deploy/kubernetes/images/prow-gke-staging-release-rc-1-19` directory with appropriate image versions. If the images used in `stable-master` or `stable-1-18` are compatible with 1.19, we can clone  from either `deploy/kubernetes/images/stable-master/image.yaml` or `deploy/kubernetes/images/stable-1-18/image.yaml`
+   * Create a new `deploy/kubernetes/overlays/prow-gke-staging-release-rc-1-19` directory and clone contents from either `deploy/kubernetes/overlays/stable-master/*` or `deploy/kubernetes/overlays/stable-1-18/*`, depending on which manifests would be compatible with the 1.19 driver bundle and add the necessary kustomize diff patches to `deploy/kubernetes/overlays/stable-1-19/*`. At this stage we want to first validate the `prow-gke-release-staging-rc-1-19` driver bundle, so care must be taken to avoid directly making changes to `deploy/kubernetes/base` manifests, as they may impact existing overlays.
```

Comment:
add "at this stage" after "avoid directly making changes to 'deploy/kubernetes/base'" - it seems like we won't always avoid it (given the options listed below).

Reply:
done